apps/api/v1/pages/api/bookings/_post.ts

```diff
@@ -218,7 +218,15 @@ async function handler(req: NextApiRequest) {
     req.userId = requestedUserId || userId;
   }
 
-  return await handleNewBooking(req, getBookingDataSchemaForApi);
+  try {
+    return await handleNewBooking(req, getBookingDataSchemaForApi);
+  } catch (error) {
+    if (error.message === ErrorCode.NoAvailableUsersFound) {
```

Comment:
We currently throw `HttpError` in some cases inside of `handleNewBooking` but that's not the correct place to throw that type of error since it's in a core library. Hence, I left it throwing an Error with the `NoAvailableUsersFound` property and now in the API layer (which handles HTTP), we set its `statusCode` to 400.